feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/internal/CachedToggleStore.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.feature.toggles.api.internal
+
+import com.duckduckgo.feature.toggles.api.Toggle
+import com.duckduckgo.feature.toggles.api.Toggle.State
+import com.google.common.cache.CacheBuilder
+import com.google.common.cache.CacheLoader
+import com.google.common.cache.LoadingCache
+import org.jetbrains.annotations.TestOnly
+import java.util.Optional
+import kotlin.jvm.optionals.getOrNull
+
+class CachedToggleStore constructor(
+    private val store: Toggle.Store,
+) : Toggle.Store {
+    @Volatile
+    private var listener: Listener? = null
```

Comment:
Having a single listener dones't allow for multiple observers, even when the API docs say: _"Multiple collectors will each register their own listener instance."_